nexus-watcher/src/events/mod.rs

```diff
@@ -91,16 +95,32 @@ impl Event {
         }))
     }
 
-    pub async fn handle(self, moderation: Arc<Moderation>) -> Result<(), DynError> {
+    pub async fn handle(&self, moderation: Arc<Moderation>) -> Result<(), DynError> {
         match self.event_type {
-            EventType::Put => self.handle_put_event(moderation).await,
-            EventType::Del => self.handle_del_event().await,
-        }
+            EventType::Put => self.handle_put_event(moderation).await?,
+            EventType::Del => self.handle_del_event().await?,
+        };
+
+        self.store_event().await
+    }
+
+    /// Stores event at redis as a member of stored set.
+    /// Sorting is done by synchronization timestamp.
+    async fn store_event(&self) -> Result<(), DynError> {
+        // We use sync timestamp instead of homeserver cursor (creation timestamp)
+        // because we want to ensure that events are processed in the order they
+        // were received and they can not be inserted in between already processed events
+        let ts_ms = SystemTime::now().duration_since(UNIX_EPOCH)?.as_millis() as f64;
+        let line = format!("{} {}", self.event_type, self.uri);
+
+        let elements = vec![(ts_ms, line.as_str())];
+
+        Event::put_index_sorted_set(&["Events"], &elements, None, None).await
```

Comment:
Are events only inserted?

Or is there some condition where entries will be deleted? For example after some time, or after they've been checked by moderation.